lib/comm_qmp.cpp

```diff
@@ -35,7 +35,33 @@ void comm_init(int ndim, const int *dims, QudaCommsMap rank_from_coords, void *m
   Topology *topo = comm_create_topology(ndim, dims, rank_from_coords, map_data);
   comm_set_default_topology(topo);
 
-  // determine which GPU this process will use (FIXME: adopt the scheme in comm_mpi.cpp)
+  // determine which GPU this rank will use
+  char *hostname = comm_hostname();
+  char *hostname_recv_buf = (char *)safe_malloc(128*comm_size());
+
+  // Abuse reductions to emulate all-gather.  We need to copy the
+  // local hostname to all other nodes
+  for (int i=0; i<comm_size(); i++) {
+    int data[128];
+    for (int j=0; j<128; j++) {
+      data[j] = (i == comm_rank()) ? hostname[j] : 0;
+    }
+    // check nasty int to float hack is valid
+    if (sizeof(float) != sizeof(int)) errorQuda("Broken");
```

Comment:
Ok, I am not familiar with QMP internals so that hack is probably necessary. But the error message should be slightly more verbose than broken. Also, not for 0.7.2: Can we catch that at compile time? Assert?